tracing/src/macros.rs

```diff
@@ -884,9 +884,63 @@ macro_rules! event {
 /// }
 /// ```
 ///
+/// # Alternatives
+///
+/// `enabled!` queries subscribers with [`Metadata`] where
+/// [`is_event`] and [`is_span`] both return `false`. Alternatively,
+/// use [`enabled_event`] or [`enabled_span`] to ensure one of these
+/// returns true.
+///
+///
+/// [`Metadata`]: crate::Metadata
+/// [`is_event`]: crate::Metadata::is_event
+/// [`is_span`]: crate::Metadata::is_span
+/// [`enabled_event`]: crate::enabled_event
+/// [`enabled_span`]: crate::enabled_span
+///
 #[macro_export]
 macro_rules! enabled {
-    (target: $target:expr, $lvl:expr, { $($fields:tt)* } )=> ({
+    ($($rest:tt)*)=> (
+        $crate::_enabled!(kind: { $crate::metadata::Kind::HINT }, $($rest)*)
+    )
+}
+
+/// The same as [`enabled`], but queries subscribers with `Metadata` where
+/// [`Metadata::is_event`] returns true.
+///
+/// See also [`enabled_span`].
+///
+/// [`enabled`]: crate::enabled
+/// [`Metadata::is_event`]: crate::Metadata::is_event
+/// [`enabled_span`]: crate::enabled_span
+///
+#[macro_export]
+macro_rules! enabled_event {
+    ($($rest:tt)*)=> (
+        $crate::_enabled!(kind: $crate::metadata::Kind::EVENT.hint(), $($rest)*)
+    )
+}
+
+/// The same as [`enabled`], but queries subscribers with `Metadata` where
+/// [`Metadata::is_span`] returns true.
+///
+/// See also [`enabled_event`].
+///
+/// [`enabled`]: crate::enabled
+/// [`Metadata::is_span`]: crate::Metadata::is_span
+/// [`enabled_event`]: crate::enabled_event
+///
+#[macro_export]
+macro_rules! enabled_span {
```

Comment:
similarly, i have a slight preference for `enabled_span!`, as i discussed on the event version of the macro.

Reply:
do you mean `span_enabled!`?